File: android/src/main/kotlin/com/microsoft/clarity/api/ClarityDeviceHostApiImpl.kt
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

package com.microsoft.clarity.api

import android.content.Context
import com.microsoft.clarity.bridges.DeviceBridge
import com.microsoft.clarity.generated.ClarityDeviceHostApi
import com.microsoft.clarity.generated.DeviceInfoData
import com.microsoft.clarity.generated.PackageInfoData

internal class ClarityDeviceHostApiImpl(
    private val context: Context,
) : ClarityDeviceHostApi {
    override fun getDeviceInfo(): DeviceInfoData = DeviceBridge.getDeviceInfo(context)

    override fun getPackageInfo(): PackageInfoData = DeviceBridge.getPackageInfo(context)

    override fun getCacheDirectory(): String? = DeviceBridge.getCacheDirectory(context)

    override fun getUserAgent(): String = DeviceBridge.getUserAgent(context)
}

File: ios/clarity_flutter/Sources/clarity_flutter/api/ClarityDeviceHostApiImpl.swift
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

internal class ClarityDeviceHostApiImpl: ClarityDeviceHostApi {

    func getDeviceInfo() throws -> DeviceInfoData { DeviceBridge.getDeviceInfo() }

    func getPackageInfo() throws -> PackageInfoData { DeviceBridge.getPackageInfo() }

    func getCacheDirectory() throws -> String? { DeviceBridge.getCacheDirectory() }

    func getUserAgent() throws -> String { DeviceBridge.getUserAgent() }
}
